src/main/java/org/folio/service/RoutingListsService.java

```diff
@@ -0,0 +1,109 @@
+
+  private final RestClient restClient;
+  private final UserService userService;
+
+  public RoutingListsService(RestClient restClient, UserService userService) {
+    this.restClient = restClient;
+    this.userService = userService;
+  }
+
+  public Future<JsonObject> processTemplateRequest(String routingListId, RequestContext requestContext) {
+    log.debug("processTemplateRequest: Tying to process template request for routingListId={}", routingListId);
+    return getRoutingListById(routingListId, requestContext)
+      .compose(routingList -> fetchUsersAndCreateTemplate(routingList, requestContext))
+      .compose(templateProcessingRequest -> postTemplateRequest(templateProcessingRequest, requestContext));
+  }
+
+  public Future<RoutingList> getRoutingListById(String routingListId, RequestContext requestContext) {
+    var requestEntry = new RequestEntry(ROUTING_LIST_BY_ID_ENDPOINT).withId(routingListId);
+    return restClient.get(requestEntry, RoutingList.class, requestContext);
+  }
+
+  private Future<TemplateProcessingRequest> fetchUsersAndCreateTemplate(RoutingList routingList, RequestContext requestContext) {
+    return userService.getUsersByIds(routingList.getUserIds(), requestContext)
+      .map(users -> createTemplateRequest(routingList, users));
+  }
+
+  private TemplateProcessingRequest createTemplateRequest(RoutingList routingList, UserCollection users) {
+    var templateRequest = createBaseTemplateRequest();
+    var usersForContext = createUsersForContext(users);
+    var routingListForContext = createRoutingListForContext(routingList);
+    templateRequest.withContext(new TemplateProcessingRequest.Context()
+        .withRoutingList(routingListForContext)
+        .withUsers(usersForContext));
+    log.info("createTemplateRequest:: TemplateProcessingRequest object created for routing list name: {}",
+      templateRequest.getContext().getRoutingList().getName());
+    return templateRequest;
+  }
+
+  private TemplateProcessingRequest createBaseTemplateRequest() {
+    return new TemplateProcessingRequest()
+      .withTemplateId(TEMPLATE_REQUEST_ID)
+      .withLang(TEMPLATE_REQUEST_LANG)
+      .withOutputFormat(TEMPLATE_REQUEST_OUTPUT);
+  }
+
+  private List<TemplateProcessingRequest.User> createUsersForContext(UserCollection userCollection) {
+    if (userCollection.getUsers().isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    return userCollection.getUsers().stream()
+      .map(UserCollection.User::getPersonal)
+      .filter(Objects::nonNull)
+      .map(personalData -> {
+          var userForContext = new TemplateProcessingRequest.User()
+            .withFirstName(personalData.getFirstName())
+            .withLastName(personalData.getLastName());
+          var addresses = personalData.getAddresses();
+          if (addresses != null && !addresses.isEmpty()){
+            userForContext.withRoutingAddress(addresses.get(0).getAddressLine1());
```

Comment:
we could not get addresses.get(0). Logic should be the following: in orders-storage we have settings API where user can select preferred address type: https://folio-org.atlassian.net/browse/UIREC-300
So here we need to iterate through addresses and choose that address where address type is matched with settings value.

Reply:
this part is on hold for now until it will be solved